client/gutenberg/editor/controller.js

```diff
@@ -98,6 +143,7 @@ function getPressThisData( query ) {
 export const post = ( context, next ) => {
 	// See post-editor/controller.js for reference.
 
+	handleJetpackSSO( context );
```

Comment:
If `handleJetpackSSO` does a redirect, you don't want to continue handling the route. All the code that follows will just do useless work that will be thrown away on the next event loop tick, when the navigation will be actually performed.

Reply:
Moved this function to an independent `page` controller in 136a57d3a6c74ae86bdbaa9e2f87ba4f8ff24149. The new controller won't execute the next callback registered in the route unless the WP Admin request is already authenticated.